common/src/main/java/com/sequenceiq/cloudbreak/auth/ThreadBasedUserCrnProvider.java

```diff
@@ -33,27 +29,18 @@ public String getAccountId() {
         }
     }
 
-    public void setUserCrn(String userCrn) {
-        Stack<String> stack = USER_CRN.get();
-        if (stack == null) {
-            stack = new Stack<>();
-            USER_CRN.set(stack);
-        }
-        if (!stack.isEmpty() && !stack.peek().equals(userCrn)) {
-            LOGGER.error("Trying to push crn to stack {} when it already contains {}", userCrn, stack.get(0));
-            // REMOVE IT UNTIL WE FIX
-            // throw new IllegalStateException(String.format("Trying to push crn to stack %s when it already contains %s", userCrn, stack.get(0)));
+    public static void setUserCrn(String userCrn) {
+        if (USER_CRN.get() != null) {
+            String errorMessage = String.format("Trying to set crn %s when it already contains %s, please check where we didn't remove it",
```

Comment:
We can print out the calltrace here so would be easy to detect

Reply:
good idea, thanks